src/structures/paging/mapper/mapped_page_table.rs

```diff
@@ -370,7 +370,7 @@ impl<'a, P: PhysToVirt> MapperAllSizes for MappedPageTable<'a, P> {
             Err(PageTableWalkError::NotMapped) => return TranslateResult::PageNotMapped,
             Err(PageTableWalkError::MappedToHugePage) => {
                 let frame = PhysFrame::containing_address(p3[addr.p3_index()].addr());
-                let offset = addr.as_u64() & 0o_777_777_7777;
+                let offset = addr.as_u64() & 0o7_777_777_777;
```

Comment:
The digits grouping was deliberate as it resembles the page table layout. Each `777` block corresponds to a 9-bit page table index and the `7777` block at the end corresponds to the 12-bit page offset. So I would prefer keeping the digits grouped as is. This also applies to the other instances of this commit.

I think it should be possible to disable this clippy lint for these functions by setting it to `allow`.